File: backend/src/main/java/edu/cit/garol/campus/claimrequest/ClaimRequest.java
package edu.cit.garol.campus.claimrequest;

import edu.cit.garol.campus.itemreport.ItemReport;
import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "claim_requests")
public class ClaimRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String claimantId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "item_report_id",
            nullable = false
    )
    private ItemReport itemReport;
    private String claimantName;
    private String claimantEmail;
    private String claimantPhone;
    private String proofImagePath;


    @Column(length = 1500)
    private String itemDescription;
    @Column(length = 1500)
    private String additionalInformation;
    private String status = "Pending";
    private LocalDateTime createdAt;

    public ClaimRequest() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getClaimantId() {
        return claimantId;
    }

    public void setClaimantId(String claimantId) {
        this.claimantId = claimantId;
    }

    public ItemReport getItemReport() {
        return itemReport;
    }

    public void setItemReport(ItemReport itemReport) {
        this.itemReport = itemReport;
    }

    public String getClaimantName() {
        return claimantName;
    }

    public void setClaimantName(String claimantName) {
        this.claimantName = claimantName;
    }

    public String getClaimantEmail() {
        return claimantEmail;
    }

    public void setClaimantEmail(String claimantEmail) {
        this.claimantEmail = claimantEmail;
    }

    public String getClaimantPhone() {
        return claimantPhone;
    }

    public void setClaimantPhone(String claimantPhone) {
        this.claimantPhone = claimantPhone;
    }

    public String getProofImagePath() {
        return proofImagePath;
    }

    public void setProofImagePath(String proofImagePath) {
        this.proofImagePath = proofImagePath;
    }

    public String getItemDescription() {
        return itemDescription;
    }

    public void setItemDescription(String itemDescription) {
        this.itemDescription = itemDescription;
    }

    public String getAdditionalInformation() {
        return additionalInformation;
    }

    public void setAdditionalInformation(String additionalInformation) {
        this.additionalInformation = additionalInformation;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: mobile/app/src/main/java/cit/edu/garol/campus/features/claimrequest/model/ClaimRequest.kt
package cit.edu.garol.campus.features.claimrequest.model

import com.google.gson.annotations.SerializedName

data class ClaimRequest(

    @SerializedName("id")
    val id: Int,

    @SerializedName("claimantId")
    val claimantId: String,

    @SerializedName("claimantName")
    val claimantName: String,

    @SerializedName("claimantEmail")
    val claimantEmail: String,

    @SerializedName("claimantPhone")
    val claimantPhone: String,

    @SerializedName("proofImagePath")
    val proofImagePath: String?,

    @SerializedName("itemDescription")
    val itemDescription: String,

    @SerializedName("additionalInformation")
    val additionalInformation: String?,

    @SerializedName("status")
    val status: String,

    @SerializedName("createdAt")
    val createdAt: String,

    @SerializedName("itemReport")
    val itemReport: ClaimRequestItemReport
)